### PR TITLE
Clock: Store time and timezone in AsyncLocal<> instead of ThreadStatic.

### PR DESCRIPTION
**Pull request by Sjaaky**
Make Clock also work with async/await.
Solves #303

### CODE REVIEW

**Comment by Corniel** [comment]
@Sjaaky What happens now if multiple tests run in parallel with different using blocks?

**Review by Corniel — APPROVED** [review]
LGTM.

File: src/Qowaiv/Qowaiv.csproj [hunk 1]
```diff
@@ -5,8 +5,14 @@
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0;net5.0;net6.0;net7.0</TargetFrameworks>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <Version>6.4.0</Version>
+    <Version>6.4.1</Version>
     <PackageReleaseNotes>
+v6.4.1
+- Clock.SetTimeForCurrentContext supports async. #304
+- Fix issue with not closed comment in parsing email adresses. #298
```
Comment (LauraXiulan) [inline src/Qowaiv/Qowaiv.csproj:12]: Are all these changes in this PR, or are they previous changes that didn't get a version bump yet?

Comment (Corniel) [reply src/Qowaiv/Qowaiv.csproj:12]: Yep. They where considered too minor to get a separate release.

File: src/Qowaiv/Clock.cs [hunk 2]
```diff
@@ -36,14 +38,14 @@ public static class Clock
...
-        var utcNow = (threadUtcNow ?? globalUtcNow).Invoke();
+        var utcNow = (threadUtcNow.Value ?? globalUtcNow).Invoke();
         return utcNow.Kind == DateTimeKind.Utc
             ? utcNow
             : new DateTime(utcNow.Ticks, DateTimeKind.Utc);
     }
 
     /// <summary>Gets the time zone of the <see cref="Clock"/>.</summary>
-    public static TimeZoneInfo TimeZone => threadTimeZone ?? globalTimeZone;
+    public static TimeZoneInfo TimeZone => threadTimeZone.Value ?? globalTimeZone;
```
Comment (LauraXiulan) [inline src/Qowaiv/Clock.cs:48]: Can threadTimeZone be null?

Comment (Corniel) [reply src/Qowaiv/Clock.cs:48]: Yep. It can. I previously was a `Func` that could be null. Now it is a `LocalAsync<T>` where the `Value` can be null.

**Review by LauraXiulan — APPROVED** [review]

**Pull request merged**